### DISCUSSION TITLE
saving a sequence in igs.mha

### DISCUSSION THREAD

**Discussion by MRezaEffatparvar** [PlusToolkit/PlusLib#1217] [Q&A] [answered]
Hello everyone,
I am trying to prepare a configuration file to replay recorded data from a sequence file that I have already saved. Based on the example configuration file, the recorded sequence should be in .igs.mha format. However, I don't see this option when saving my sequence. The available options are .mha, .mhd, .seq.mha, and .seq.mhd.

From the documentation, I understand that I may need the IGSIO extension to enable the .igs.mha format. However, when I install IGSIO, it only adds a module called "VideoUtil" and doesn't seem to modify the available saving options. Could you help me resolve this issue?

Thanks!

**Comment by lassoan** [comment]
If you record sequences in Slicer then you can replay them in Slicer. If you need to send the replayed data via OpenIGTLink then Slicer can do it, too.

Slicer can also read .igs.mha, .igs.mhd, .igs.nrrd files (see [qSlicerSequenceIOReader](https://github.com/IGSIO/SlicerIGSIO/blob/master/SequenceIO/qSlicerSequenceIOReader.cxx)), but there is no exporter. Writing an exporter is not trivial, because usually an .igs.mha/mhd/nrrd file contains multiple sequences, which are loaded into Slicer as multiple nodes, so in an exporter you would need to merge these multiple sequences into a single file.

You can use `Plus Remote` module in Slicer to remote control your PLUS server to start/stop recording data. It saves data in .igs.mha/mhd/nrrd files that can be replayed in PLUS or loaded into Slicer.

The only conversion that is missing is that if you import .igs.mha/mhd/nrrd into Slicer then you cannot export it as .igs.mha/mhd/nrrd from Slicer.

**Reply by MRezaEffatparvar** [reply]
Thanks for your reply.
What I am trying to do is, after recording my sequence and all related data, send them to another server. Then, I will receive this data using PyIGTL, run AI segmentation on it, and send it back to Slicer through another server.

May I ask what options I have for sending the recorded sequence? As a test, in OpenIGTLinkIF, I created a server to send the sequence. Then, I opened another instance of 3D Slicer and tried to open a new client with the same listener port. However, I only get an OpenIGTLink connection without seeing additional information, such as the image or transform.

**Reply by MRezaEffatparvar** [reply]
Yes, I just read about the virtual capturing option in the configuration file for recording sequences, and this is great—I will definitely use it for my next recordings. However, I have already recorded some sequences in 3D Slicer, and I want to work on them.

Do I have any options for sending this recorded data to another server directly from Slicer? The sequences are in .mha format.

**Reply by lassoan** [reply]
> May I ask what options I have for sending the recorded sequence?

In OpenIGTLink module, you need to add all transform nodes that you want to send to the list of outgoing nodes.

**Reply by MRezaEffatparvar** [reply]
Great, I can send the transforms. Is there any way to send the sequence through this link as well? The issue is that the image and transform are not updating on the other side because the sequence is missing.

Thanks again!

**Reply by lassoan** [reply, marked as answer]
You always send the current state of the node. If you want to record it into a sequence then in Sequences module you can create a sequence, assign the node that OpenIGTLink updates as proxy node, and start recording.

**Reply by MRezaEffatparvar** [reply]
Thank you so much
